src/api/app/controllers/distributions_controller.rb

```diff
@@ -55,17 +56,9 @@ def upload
   # POST /distributions
   # POST /distributions.xml
   def create
-    @distribution = Distribution.new(params[:distribution])
+    Distribution.parse(Xmlhash.parse(request.body.read), delete_current: false)
```

Comment:
I don't think the method parse should call `create`. Maybe instead return 'Distribution#new' and then you try to save in this controller and keep the `respond_to` block the same?

Reply:
So this was actually kind of the first thing I proposed. @dmarcoux suggested this approach as the point of this PR isn't to refactor the code, but instead to reuse what is already present. The whole controller needs to be refactored anyway in the future and this was meant to just to get the POST endpoint working while modifying the existing code for other methods as little as possible.

Reply:
> but instead to reuse what is already present

Therefore IMO, if you have the `Distribution.parse` returning `Distribution.new()`, you don't have to change anything on the controller.

Reply:
In the controller, yes. But it would require more changes to the model to do it this way without changing the behaviour of PUT.